src/execution/join_hashtable.cpp

```diff
@@ -57,11 +60,55 @@ JoinHashTable::JoinHashTable(BufferManager &buffer_manager, const vector<JoinCon
 	idx_t block_capacity = MaxValue<idx_t>(STANDARD_VECTOR_SIZE, (Storage::BLOCK_SIZE / entry_size) + 1);
 	block_collection = make_unique<RowDataCollection>(buffer_manager, block_capacity, entry_size);
 	string_heap = make_unique<RowDataCollection>(buffer_manager, (idx_t)Storage::BLOCK_SIZE, 1, true);
+	swizzled_block_collection = block_collection->CloneEmpty();
+	swizzled_string_heap = string_heap->CloneEmpty();
 }
 
 JoinHashTable::~JoinHashTable() {
 }
 
+void JoinHashTable::Merge(JoinHashTable &other) {
+	block_collection->Merge(*other.block_collection);
+	swizzled_block_collection->Merge(*other.swizzled_block_collection);
+	if (!layout.AllConstant()) {
+		string_heap->Merge(*other.string_heap);
+		swizzled_string_heap->Merge(*other.swizzled_string_heap);
+	}
+
+	if (join_type == JoinType::MARK) {
+		auto &info = correlated_mark_join_info;
+		lock_guard<mutex> mj_lock(info.mj_lock);
+		has_null = has_null || other.has_null;
+		if (!correlated_mark_join_info.correlated_types.empty()) {
```

Comment:
The `correlated_types` field should be the same for `this` and `other`, because it gets set on initialization of the HT, so no need to copy the vector holding the types.

Only the aggregated data in `correlated_counts` needs to be combined (and the `has_null` flag needs to be correctly set).